src/scheduler/pbs_sched_utils.cpp

```diff
@@ -629,19 +629,14 @@ connect_svrpool()
 			goto unmask_continue;
 		}
 
-		/* Reached here means everything is success, so we will break out of the loop */
-unmask_break:
-		if (sigprocmask(SIG_SETMASK, &prevsigs, NULL) == -1)
-			log_err(errno, __func__, "sigprocmask(SIG_SETMASK)");
-		break;
```

Comment:
Thanks for fixing this issue.
I don't think this change is needed. Before I changed it, it was breaking out of the outer for loop here. If you remove the break here then the while loop can potentially become endless.

Reply:
oops, thanks for catching this, we need to add the break back.

Reply:
Also, retain the sigprocmask() call.

Reply:
If you don't add back the sigprocmask call, the signals will stay blocked.